core/chain_makers.go

```diff
@@ -420,9 +438,23 @@ func GenerateVerkleChain(config *params.ChainConfig, parent *types.Block, engine
 			proofs = append(proofs, block.ExecutionWitness().VerkleProof)
 			keyvals = append(keyvals, block.ExecutionWitness().StateDiff)
 
-			return block, b.receipts
+			computedPostStateValues := make([][]byte, len(computedKeys))
+			treeWrites := statedb.GetTrie().(*trie.VerkleTrie).GetTreeWrites()
+			for i := range computedKeys {
+				treeWrittenValue, ok := treeWrites[string(computedKeys[i])]
+				// If we didn't tracked this key, it means it was never written to the post-state tree,
+				// thus the newValue must be `nil`. (i.e: same as currentValue)
+				// Additionally, if we wrote to this key but it has the same pre-state value, then must
+				// also be `nil`.
+				if !ok || bytes.Equal(treeWrittenValue, computedPreStateValues[i]) {
+					continue
+				}
+				computedPostStateValues[i] = treeWrittenValue
+			}
```

Comment:
Note that here, we call a new `GetTreeWrites()` from the tree.
This will return all the detected tree writes as a `map[string][]byte` (i.e: map[treeKey]writtenTreeValue).

Now, remember that just because some key was written in the tree doesn't mean that it actually mutated the tree (what I explained in Matrix). To solve that, we use both `computedKeys` and `computedPreStateValues` to construct the expected post-values list.

Note that this is safe, since both `computedKeys` and `computedPreStateValues` is data that the node has calculated itself (i.e: **it is not** relying on block proof stuff, which isn't trusted).